modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -0,0 +1,348 @@
+
+        if(attribute_name == "kernel_shape")
+        {
+            CV_Assert(attribute_proto.ints_size() == 2);
+            lp.set("kernel_h",  attribute_proto.ints(0));
+            lp.set("kernel_w",  attribute_proto.ints(1));
+        }
+        else if(attribute_name == "strides")
+        {
+            CV_Assert(attribute_proto.ints_size() == 2);
+            lp.set("stride_h",  attribute_proto.ints(0));
+            lp.set("stride_w",  attribute_proto.ints(1));
+        }
+        else if(attribute_name == "pads")
+        {
+            CV_Assert(attribute_proto.ints_size() >= 2);
+            lp.set("pad_h", attribute_proto.ints(0));
+            lp.set("pad_w", attribute_proto.ints(1));
+        }
+        else if (attribute_proto.has_i())
+        {
+            lp.set(attribute_name, attribute_proto.i());
+        }
+        else if (attribute_proto.has_f())
+        {
+            lp.set(attribute_name, attribute_proto.f());
+        }
+        else if (attribute_proto.has_s())
+        {
+            lp.set(attribute_name, attribute_proto.s());
+        }
+        else if (attribute_proto.floats_size() > 0)
+        {
+            lp.set(attribute_name, DictValue::arrayReal(
+                (float*)attribute_proto.mutable_floats(), attribute_proto.floats_size()));
+        }
+        else if (attribute_proto.ints_size() > 0)
+        {
+                lp.set(attribute_proto.name(), DictValue::arrayInt(
+                    (int*)attribute_proto.mutable_ints(), attribute_proto.ints_size()));
+        }
+        else if (attribute_proto.has_t() || attribute_proto.has_g() ||
+                    attribute_proto.strings_size() > 0 ||
+                        attribute_proto.tensors_size() > 0 ||
+                            attribute_proto.graphs_size() > 0)
+        {
+                CV_Error(Error::StsNotImplemented, "Unexpected attribute type");
+        }
+    }
+    return lp;
+}
+
+Mat ONNXImporter::getBlob(const opencv_onnx::NodeProto& node_proto,
+                    const std::map<std::string, Mat>& constBlobs, int index)
+{
+    std::map<std::string, Mat>::const_iterator constBlob;
+    constBlob = constBlobs.find(node_proto.input(index) );
+    if (constBlob == constBlobs.end()) {
+        CV_Error(Error::StsObjectNotFound,
+             "Blob " + node_proto.input(index) + " not found in const blobs");
+    }
+    return constBlob->second;
+}
+
+void ONNXImporter::populateNet(Net dstNet)
+{
+    CV_Assert(model_proto.has_graph());
+    opencv_onnx::GraphProto graph_proto = model_proto.graph();
+    std::map<std::string, Mat> constBlobs = getGraphTensors(graph_proto);
+
+    std::string framework_name;
+    if (model_proto.has_producer_name()) {
+        framework_name = model_proto.producer_name();
+    }
+
+    std::map<std::string, int> layer_id;
+    std::map<std::string, int>::iterator layerId, currentId;
+
+    int count = 0;
+    for (int j = 0; j < graph_proto.input_size(); j++)
```

Comment:
Please add some comment that you register network's inputs here.

Reply:
done